File: barryma_rapport_stage_1/project_covid19_final/scripts/spark/python/DataAggregation.py
from pyspark.sql import SparkSession
from pyspark.sql.functions import sum, avg, max, min

# Create a SparkSession
spark = SparkSession.builder.appName("Data Aggregation").getOrCreate()

# Load the dataset
df = spark.read.option("header", "true").csv(sys.argv[1])

# Example aggregation: Group by a column and calculate multiple aggregates
aggregatedDF = df.groupBy("country") \
  .agg(
    sum("cases").alias("total_cases"),
    avg("cases").alias("avg_cases"),
    max("cases").alias("max_cases"),
    min("cases").alias("min_cases")
  )

# Save the aggregated dataset
aggregatedDF.write.option("header", "true").csv(sys.argv[2])

spark.stop()

File: barryma_rapport_stage_1/project_covid19_final/scripts/spark/scala/DataAggregation.scala
import org.apache.spark.sql.SparkSession
import org.apache.spark.sql.functions._

object DataAggregation {
  def main(args: Array[String]): Unit = {
    val spark = SparkSession.builder.appName("Data Aggregation").getOrCreate()
    
    // Load the dataset
    val df = spark.read.option("header", "true").csv(args(0))
    
    // Example aggregation: Group by a column and calculate multiple aggregates
    val aggregatedDF = df.groupBy("country")
      .agg(
        sum("cases").as("total_cases"),
        avg("cases").as("avg_cases"),
        max("cases").as("max_cases"),
        min("cases").as("min_cases")
      )
    
    // Save the aggregated dataset
    aggregatedDF.write.option("header", "true").csv(args(1))
    
    spark.stop()
  }
}
